numbers/monteCarlo.go

```diff
@@ -89,25 +89,38 @@ func chooseBin(sumHeights float64, binHeights []float64) int {
 }
 
 //RejectionSample returns simulated values from an arbitrary function between a specified left and right bound using a simple rejection sampling method.
-func RejectionSample(xLeft float64, xRight float64, yMax float64, f func(float64) float64, maxIteration int, returnX bool) float64 {
+func RejectionSample(xLeft float64, xRight float64, yMax float64, f func(float64) float64, maxIteration int) float64 {
 	var x, y float64
 	for i := 0; i < maxIteration; i++ {
 		x = RandFloat64InRange(xLeft, xRight) //rand float64 in range xleft to xright
 		y = f(x)
 		if RandFloat64InRange(0.0, yMax) < y {
-			if returnX {
-				return x
-			}
 			return y
 		}
 	}
 	log.Fatalf("Exceeded max iterations.")
 	return -1.0
 }
 
+func BoundedRejectionSample(boundingSampler func() (float64, float64), f func(float64) float64, xLeft float64, xRight float64, maxIteration int) float64 {
+	var xSampler, ySampler, y float64
+	for i := 0; i < maxIteration; i++ {
+		xSampler, ySampler = boundingSampler()
+		y = f(xSampler)
+		if y > ySampler {
+			log.Fatalf("BoundedRejectionSample: function was not a valid bounding function, ySampler is greater than y.")
+		}
+		if RandFloat64InRange(0.0, ySampler) < y {
+			return y
```

Comment:
Should this be returning xSampler?

Reply:
yes thanks, my mistake. In fact, it may be useful to return the x y point